test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/PushCommandTest.cs

```diff
@@ -245,6 +256,512 @@ public void PushCommand_Server_SkipDuplicate_IsSpecified_PushProceeds()
+
+                    // Act
+                    var result = CommandRunner.Run(
+                        nuget,
+                        packageDirectory,
+                        $"push {nupkgFullPath} -Source {sourceName} -Timeout 110",
+                        waitForExit: true,
+                        timeOutInMilliseconds: 120000); // 120 seconds
+
+                    // Assert
+
+                    //Ignoring filename in File Not Found error since the error should not appear in any case.
+                    string genericFileNotFoundError = string.Format(MESSAGE_FILE_DOES_NOT_EXIST, string.Empty);
+
+                    Assert.True(0 == result.Item1, "Expected to successfully push a nupkg without a snupkg.");
+                    Assert.Contains(MESSAGE_PACKAGE_PUSHED, result.Item2);
+                    Assert.DoesNotContain(genericFileNotFoundError, result.Item3);
+                }
+            }
+        }
+
+        /// <summary>
+        /// When pushing *.nupkg to a symbol server, but no snupkgs are selected with that wildcard, there is (TODO: not) a FileNotFound error about snupkgs.
+        /// TODO: bug fixes will come from https://github.com/NuGet/Home/issues/8148
+        /// </summary>
+        [Fact]
+        public void PushCommand_Server_Nupkg_WildcardSnupkgDoesNotExist_FileNotFoundError()
+        {
+            // Arrange
+            using (var packageDirectory = TestDirectory.Create())
+            {
+                var nuget = Util.GetNuGetExePath();
+                string pushArgument = "*.nupkg";
+
+                using (var server = CreateAndStartMockV3Server(packageDirectory, out string sourceName))
+                {
+                    // Act
+                    var result = CommandRunner.Run(
+                        nuget,
+                        packageDirectory,
+                        $"push {pushArgument} -Source {sourceName} -Timeout 110",
+                        waitForExit: true,
+                        timeOutInMilliseconds: 120000); // 120 seconds
+
+                    // Assert
+                    server.Stop();
+
+                    string expectedFileNotFoundErrorMessage = string.Format(MESSAGE_FILE_DOES_NOT_EXIST, pushArgument);
+
+                    Assert.False(0 == result.Item1, "File did not exist and should fail.");
+                    Assert.DoesNotContain(MESSAGE_PACKAGE_PUSHED, result.Item2);
+
+                    //TODO: Will become DoesNotContain after bug fixes.
+                    Assert.Contains(expectedFileNotFoundErrorMessage, result.Item3);
+                }
+            }
+        }
+
+        /// <summary>
+        /// When pushing a nupkg by filename to a Symbol Server with a matching snupkg, a 409 Conflict halts the push.
+        /// TODO: bug fixes will come from https://github.com/NuGet/Home/issues/8148
+        /// </summary>
+        [Fact]
+        public void PushCommand_Server_Nupkg_FilenameSnupkgExists_Conflict()
```

Comment:
The name doesn't really communicate the intent here clearly. 

Is the snupkg push failing because the equivalent nupkg push did?

Reply:
How is this different from the below test?

Reply:
The pattern is PushCommand_Server_<what is being pushed>_<filename/wildcard><does the file exist?>_<Server Response> and I'm open to suggested changes to the pattern, which follows similarly to the other Functional Tests.

Reply:
The difference is in bold: PushCommand_Server_Nupkg_**WildcardFindsMatching**Snupkgs_Conflict

Meaning that this test is using a Wildcard rather than a Filename, and the wildcard finds matching files, but the server returns a Conflict.

Reply:
The fact that nuget.org might 409/201 even after a push wasn't clear. 
Given that I can see why you want the 2 tests.